### ISSUE TITLE
Improve window search

### ISSUE THREAD

**Issue by tadly** [Tadly/hideIt.sh#2]
Currently only `WM_NAME` will be taken into account while identifying a window.
A unique match is not guarantied and falling back to the first match out of the whole set is likely to be wrong.

I'd like to expand the search by at least `WM_CLASS` for additional filtering.

**Comment by rotsix** [comment]
Yep, if I were you, I would add `id`  or `instance` or even `title` with a regex matching O:)

**Comment by tadly** [comment]
`title` is what will be used right now (probably should change the cli naming for that).

I'm still fiddling with `xdotool search` but it's just not doing what I'd expect it to.
e.g. open a nautilus window.
class will be `Nautilus` while instance will be `nautilus`.

Doing `xdotool search --class 'nautilus'` I'd expect *no* mach, yet I get one.

I just always get more results than I'd expect and haven't found a proper/nice way of doing it :/

**Comment by tadly** [comment]
I started digging again and it's actually a somewhat strange search implementation within xdotool which doesn't allow to combine class, instance and title searches with different search-terms.

There's an [issue](https://github.com/jordansissel/xdotool/issues/180) where I added my 2 cents. Let's see where this'll go.

Until than I'll try and work around it using multiple search requests somehow :)

**Comment by tadly** [comment]
This took way to long to implement ':D
But here we go (I hope), better search -> 665c0bd981bdfa51278becc49d9925a0208824fc

I'll still leave that one open as the implementation is just a workaround and I want to see what **xdotool** will do as he isn't happy with his implementation either :)
